### PR TITLE
Bump Oasis Core to 24.0

### PR DESCRIPTION
**Pull request by kostko**

### CODE REVIEW

File: client-sdk/ts-web/core/reflect-go/main.go [hunk 1]
```diff
@@ -563,6 +566,8 @@ var skipMethods = map[string]bool{
 	"github.com/oasisprotocol/oasis-core/go/consensus/api.ClientBackend.Governance": true,
 	"github.com/oasisprotocol/oasis-core/go/consensus/api.ClientBackend.RootHash":   true,
 	"github.com/oasisprotocol/oasis-core/go/consensus/api.ClientBackend.State":      true,
+	"github.com/oasisprotocol/oasis-core/go/keymanager/api.Backend.Secrets":         true,
+	"github.com/oasisprotocol/oasis-core/go/keymanager/api.Backend.Churp":           true,
```
Comment (pro-wh) [inline client-sdk/ts-web/core/reflect-go/main.go:570]: will need some extra visitClientWithService calls for https://github.com/oasisprotocol/oasis-core/blob/master/go/keymanager/churp/grpc.go#L15 and https://github.com/oasisprotocol/oasis-core/blob/master/go/keymanager/churp/grpc.go#L15

File: tests/run-e2e.sh [hunk 2]
```diff
@@ -98,6 +98,7 @@ printf "${CYAN}### Running end-to-end tests...${OFF}\n"
 	--e2e.runtime.binary_dir.default="${TEST_BASE_DIR}" \
 	--e2e.runtime.loader="${TEST_RUNTIME_LOADER}" \
 	--e2e.keymanager.binary="${TEST_KM_BINARY}" \
+	--scenario_timeout 30m \
```
Comment (peternose) [inline tests/run-e2e.sh:101]: Could also add context to
```go
func (sc *RuntimeScenario) waitNodesSynced() error {
```

File: runtime-sdk/src/runtime.rs [hunk 3]
```diff
@@ -50,7 +50,7 @@ pub trait Runtime {
 
     /// Return the trusted policy signers for this runtime; if `None`, a key manager connection will
     /// not be established on startup.
-    fn trusted_policy_signers() -> Option<TrustedPolicySigners> {
+    fn trusted_policy_signers() -> Option<TrustedSigners> {
```
[the remark this answers is not in the document]
Comment (kostko) [reply runtime-sdk/src/runtime.rs:53]: Good idea!

File: tests/run-e2e.sh (hunk 2 above)
Comment (kostko) [reply tests/run-e2e.sh:101]: Good idea, done.

File: tests/runtimes/simple-contracts/src/lib.rs [hunk 4]
```diff
@@ -38,8 +37,8 @@ impl sdk::Runtime for Runtime {
         contracts::Module<Config>,
     );
 
-    fn trusted_policy_signers() -> Option<TrustedPolicySigners> {
-        Some(TrustedPolicySigners::default())
+    fn trusted_policy_signers() -> Option<TrustedSigners> {
```
[the remark this answers is not in the document]
Comment (kostko) [reply tests/runtimes/simple-contracts/src/lib.rs]: Done.

File: runtime-sdk/src/runtime.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (kostko) [reply runtime-sdk/src/runtime.rs]: Done.

**Review by peternose — APPROVED** [review]

File: tests/e2e/scenario/events.go [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+		select {
+		case ev, ok := <-ch:
+			if !ok {
+				return fmt.Errorf("channel closed")
+			}
+			log.Debug("received event", "event", ev)
+			if check(ev) {
+				return nil
+			}
+		case <-time.After(timeout):
```
[the remark this answers is not in the document]
Comment (kostko) [reply tests/e2e/scenario/events.go:30]: Yeah all of these need to be refactored to take a context, but I didn't want to do it in this PR. For now they are just copied over from before.

File: client-sdk/ts-web/core/src/client.ts [hunk 6]
```diff
@@ -1115,65 +1085,13 @@ export class NodeInternal extends GRPCWrapper {
...
-        return this.callUnary(methodDescriptorKeyManagerGetStatus, query);
-    }
-
-    /**
-     * GetStatuses returns all currently tracked key manager statuses.
-     */
-    keyManagerGetStatuses(height: types.longnum) {
-        return this.callUnary(methodDescriptorKeyManagerGetStatuses, height);
-    }
-
```
Comment (lukaw3d) [inline client-sdk/ts-web/core/src/client.ts:1145]: were these intentionally removed?

Comment (kostko) [reply client-sdk/ts-web/core/src/client.ts:1145]: Yes. With CHURP the key manager actually has multiple sub-modules, "secrets" and "churp".

**Pull request merged**